src/script/descriptor.h

```diff
@@ -177,6 +178,17 @@ struct Descriptor {
  */
 std::vector<std::unique_ptr<Descriptor>> Parse(std::span<const char> descriptor, FlatSigningProvider& out, std::string& error, bool require_checksum = false);
 
+/** Convenience overload for string literals that trims the trailing null byte. */
+template <size_t N>
+std::vector<std::unique_ptr<Descriptor>> Parse(const char (&descriptor)[N], FlatSigningProvider& out, std::string& error, bool require_checksum = false)
+{
+    std::span sp{descriptor, N};
+    if (!sp.empty() && sp.back() == '\0') {
```

Comment:
An alternative approach could be to not bother with this overload and simply have `Parse` do the removal at the top.